library/spdm_common_lib/libspdm_com_crypto_service.c

```diff
@@ -882,38 +941,53 @@ bool libspdm_verify_certificate_chain_hash(libspdm_context_t *spdm_context,
             LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "!!! verify_certificate_chain_hash - FAIL !!!\n"));
             return false;
         }
+    }
```

Comment:
It seems you removed "else" logic.
Do you assume buffer_hash_size is always !=0?
If yes, I suggest to add ASSERT().

Reply:
> It seems you removed "else" logic.

Yes. The "else" logic was trying to read provisioned peer certificate chain from context.
It is no longer needed since local_context.peer_cert_chain_provision is gone.

> Do you assume buffer_hash_size is always !=0?
> If yes, I suggest to add ASSERT().

Yes, buffer_hash_size must not be 0 here.
Because this function is called in CHALLENGE process,
the certificate chain hash must be there from GET_CERTIFICATE process or imported from context cache.
I'll add the assertion.